relay-general/src/store/transactions/processor.rs

```diff
@@ -396,6 +407,45 @@ impl Processor for TransactionsProcessor<'_> {
         Ok(())
     }
 
+    /// Validates the length and characters of measurement names.
+    fn process_measurements(
+        &mut self,
+        value: &mut crate::protocol::Measurements,
+        _meta: &mut Meta,
+        _state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        let measurements = &mut value.0;
+
+        measurements.retain(|key, val| {
+            if !is_valid_metric_name(key) {
+                relay_log::error!("Measurement name contains invalid characters");
+                return false;
+            }
+
+            let name_len = key.len();
+
+            let unit_len = val
```

Comment:
which scenarios is the unit value missing from Measurement? and in those scenarios, do we want to still keep the measurement or should we change this to return false?